### PR TITLE
Unified halt being called during the cleanup

### PR DESCRIPTION
**Pull request by teclator**
- Unified the way we were halting the system. (https://bugzilla.suse.com/show_bug.cgi?id=1095251)
- Allowed to going back from the congratulate step. (https://bugzilla.suse.com/show_bug.cgi?id=1095253)
- It also depends on yast/yast-installation#716

## Finish step
Congratulate
## Go back allowed
GoingBack
## OnFailure
abortingshutdown
## OnFailure doesn't remove the reconfig_system file as expected
rebootafterabort

### CODE REVIEW

File: scripts/Firstboot-Stage/S09-cleanup [hunk 1]
```diff
@@ -49,3 +49,9 @@ if [ -e "/var/lib/YaST2/firstboot_reboot_after_finish" ] ; then
     rm /var/lib/YaST2/firstboot_reboot_after_finish
     /sbin/reboot
 fi
+
+if [ -e "/var/lib/YaST2/firstboot_halt_after_finish" ] ; then
+    log "\tHalting system as requested by firstboot..."
+    rm /var/lib/YaST2/firstboot_halt_after_finish
+    /sbin/halt -p
```
Comment (teclator) [inline scripts/Firstboot-Stage/S09-cleanup]: Probably we should not use -p

File: src/clients/firstboot_write.rb [hunk 2]
```diff
@@ -94,10 +95,6 @@ def main
         Firstboot.default_wm
       )
 
-
-      # Dont run firstboot next time
-      SCR.Execute(path(".target.remove"), Firstboot.reconfig_file)
```
Comment (jreidinger) [inline src/clients/firstboot_write.rb]: why is this needed? what happens after next reboot?

Comment (teclator) [reply src/clients/firstboot_write.rb]: It is to make the workflow consistent, if you go back from the last step, and abort the installation then the write client will have already removed the reconfig_file and you will not get into the firsboot setup.

If you finish the installation then the file will be removed by the YaST2-Firstboot.service 

https://github.com/yast/yast-installation/blob/e42d5b8cde5852e913c036102e08cf8c0c18de38/package/YaST2-Firstboot.service#L15

Comment (jreidinger) [reply src/clients/firstboot_write.rb]: ok

Comment (teclator) [reply src/clients/firstboot_write.rb]: Other option if we do not want to rely on the Firstboot.service at all for removing it is restoring the file if going back

File: src/clients/firstboot_licenses.rb [hunk 3]
```diff
@@ -71,11 +71,7 @@ def main
 
       @result = WFM.CallFunction("inst_license", [@args])
 
-      if @result == :halt
-        UI.CloseDialog
-        Builtins.y2milestone("Halting the system...")
-        SCR.Execute(path(".target.bash"), "/sbin/halt")
```
Comment (lslezak) [inline src/clients/firstboot_licenses.rb]: Um, where is this `/sbin/halt` executed now?

Comment (teclator) [reply src/clients/firstboot_licenses.rb]: It is now handled by the Service, I have unified it and now instead of each client calling halt directly halt, they just return :halt and is the `firstboot client` [here](https://github.com/yast/yast-firstboot/pull/49/files#diff-1d6918861bf68a8e44766d2cc8cad4caR112) who touch a flag file which is then checked by the S09-cleanup script for returning exit code 0 in that case

Then the YaST2-Firsboot.Service calls shutdown on failure.
